src/ObjectLock.php

```diff
@@ -621,11 +623,53 @@ public static function rawSearchOptionsToAdd($itemtype)
                 'massiveaction' => false,
                 'forcegroupby'  => true
             ];
+
+            $tab[] = [
+                'id'            => '207',
+                'table'         => getTableForItemType('ObjectLock'),
+                'field'         => 'id',
+                'datatype'      => 'specific',
+                'name'          => __('Lock status'),
+                'joinparams'    => ['jointype' => 'itemtype_item'],
+                'massiveaction' => false,
+                'forcegroupby'  => true,
+                'additionalfields' => ['date', 'users_id']
+            ];
         }
 
         return $tab;
     }
 
+    public static function getSpecificValueToDisplay($field, $values, array $options = [])
+    {
+        if (!is_array($values)) {
+            $values = [$field => $values];
+        }
+
+        switch ($field) {
+            case 'id':
+                $templateContent = <<<TWIG
+                    {% set color = is_locked ? 'bg-red-lt' : 'bg-green-lt' %}
+                    {% set icon = is_locked ? 'ti-lock' : 'ti-lock-open' %}
+                    {% set text = is_locked ? __('Locked') : __('Free') %}
+                    {% set tooltip = is_locked ? __('Locked by %s at %s')|format(call('getUsername', [users_id]), date) : text %}
```

Comment:
Call to `getUsername()` can be done on PHP side.
```suggestion
                    {% set tooltip = is_locked ? __('Locked by %s at %s')|format(user_name, date) : text %}
```